validator_client/src/initialized_validators.rs

```diff
@@ -989,7 +989,25 @@ impl InitializedValidators {
 
         let cache =
             KeyCache::open_or_create(&self.validators_dir).map_err(Error::UnableToOpenKeyCache)?;
-        let mut key_cache = self.decrypt_key_cache(cache, &mut key_stores).await?;
+
+        // Check if there is at least one local definition.
+        let has_local_definitions = self.definitions.as_slice().iter().any(|def| {
+            matches!(
+                def.signing_definition,
+                SigningDefinition::LocalKeystore { .. }
+            )
+        });
+
+        let mut key_cache;
+
+        // Only decrypt cache when there is at least one local definition.
+        // Decrypting cache is a very expensive operation which is never used for web3signer.
+        if has_local_definitions {
+            key_cache = self.decrypt_key_cache(cache, &mut key_stores).await?;
+        } else {
+            // Assign an empty KeyCache if all definitions are of the Web3Signer type.
+            key_cache = KeyCache::new();
+        }
```

Comment:
Just a small suggestion on Rustic style

```suggestion
        // Only decrypt cache when there is at least one local definition.
        // Decrypting cache is a very expensive operation which is never used for web3signer.
        let key_cache = if has_local_definitions {
            self.decrypt_key_cache(cache, &mut key_stores).await?
        } else {
            // Assign an empty KeyCache if all definitions are of the Web3Signer type.
            KeyCache::new()
        };
```

Reply:
Thank you! Fixed.